src/openrct2-ui/windows/Research.cpp

```diff
@@ -154,119 +113,248 @@ static constexpr const StringId ResearchStageNames[] = {
     STR_RESEARCH_STAGE_UNKNOWN,
 };
 
-static void WindowResearchSetPage(WindowBase* w, int32_t page);
-static void WindowResearchSetPressedTab(WindowBase* w);
-static void WindowResearchDrawTabImages(DrawPixelInfo& dpi, WindowBase* w);
-
-WindowBase* WindowResearchOpen()
+class ResearchWindow final : public Window
 {
-    WindowBase* w;
+public:
+    ResearchWindow()
+    {
+        widgets = window_research_page_widgets[WINDOW_RESEARCH_PAGE_DEVELOPMENT];
+        width = WW_DEVELOPMENT;
+        height = WH_DEVELOPMENT;
+    }
 
-    w = WindowBringToFrontByClass(WindowClass::Research);
-    if (w == nullptr)
+    void SetPage(int32_t p)
+    {
+        page = p;
+        frame_no = 0;
+        RemoveViewport();
+
+        hold_down_widgets = 0;
+        widgets = window_research_page_widgets[p];
+        disabled_widgets = 0;
+        pressed_widgets = 0;
+
+        Invalidate();
+        if (p == WINDOW_RESEARCH_PAGE_DEVELOPMENT)
+        {
+            width = WW_DEVELOPMENT;
+            height = WH_DEVELOPMENT;
+        }
+        else
+        {
+            width = WW_FUNDING;
+            height = WH_FUNDING;
+        }
+        WindowEventResizeCall(this);
+        WindowEventInvalidateCall(this);
```

Comment:
Just call the replacement member functions instead.

```suggestion
        OnResize();
        OnPrepareDraw();
```

I've got a bunch the last call is only meant to invalidate the window instead, which is (probably) already handled by the resize call. Could you try just removing it, or otherwise calling `Invalidate()` instead and see if it makes a difference?

Reply:
I have removed the `OnPrepareDraw()` call and tested. All looks fine to me.